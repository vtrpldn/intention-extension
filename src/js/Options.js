import React from 'react'
import ReactDOM from 'react-dom'
import styled from 'styled-components'

import GlobalStyle from './GlobalStyle';
import Title from './components/Title/Title'
import Button from './components/Button/Button'
import LogTable from './components/LogTable/LogTable'

class Options extends React.Component {
  constructor () {
    super()
    this.state = {
      siteList: '',
      logs: [
        {
          timestamp: 'timestamp',
          reason: 'reason',
          time: 'time'
        }
      ]
    }
  }
 
  componentDidMount () {
    chrome.storage.sync.get(['siteList', 'logs'], (items) => {
      this.setState({
        siteList: items.siteList,
        logs: items.logs
      })
    })
  }

  onChange (e) {
    this.setState({
      siteList: e.target.value
    })
  }

  saveConfig () {
    chrome.storage.sync.set({
      siteList: this.state.siteList
    }, function () {
      alert('Saved')
    })
  }

  render () {
    const Wrapper = styled.div`
      background: black;
      color: white;
      height: 100vh;
      padding: 40px 50px;
    `
    const Panel = styled.div`
      max-width: 40vw;
    `

    return (
      <Wrapper>
        <GlobalStyle />
        <Panel>
          <div>
            <Title text="Restricted sites" />
            <textarea key='sitelist' onChange={this.onChange.bind(this)} value={this.state.siteList} className='settings__field' />
          </div>
          <div>
            <Title text="Usage log" />
            <LogTable logs={this.state.logs !== undefined ? this.state.logs : []} />
          </div>
          <Button block onClick={this.saveConfig.bind(this)} text="Save all" />
        </Panel>
      </Wrapper>
    )
  }
}

const app = document.createElement('div')
document.body.appendChild(app)
ReactDOM.render(<Options />, app)
