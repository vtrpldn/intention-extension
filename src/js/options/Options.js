import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';

import LogTable from './components/LogTable/LogTable'

class Options extends React.Component {

    constructor() {
        super()
        this.state = {
            siteList: '',
            logs: [
                {
                    timestamp: 'timestamp',
                    reason: 'reason',
                    time: 'time'
                }
            ]
        }
    }

    componentDidMount() {
        chrome.storage.sync.get(['siteList', 'logs'], (items) => {
            this.setState({
                siteList: items.siteList,
                logs: items.logs
            })
        })
    }

    onChange(e) {
        this.setState({
            siteList: e.target.value
        })
    }

    saveConfig() {
        chrome.storage.sync.set({
            siteList: this.state.siteList
        }, function () {
            alert('Saved')
        })
    }

    render() {
        return (
            <div>
                <h2>Options</h2>
                <div>
                    <div>
                        Restricted sites
                    </div>
                    <textarea key="sitelist" onChange={this.onChange.bind(this)} value={this.state.siteList} className="settings__field"></textarea>
                    <button onClick={this.saveConfig.bind(this)}> Salvar </button>
                </div>
                <div>
                    <div>
                        Usage log
                    </div>
                    <LogTable logs={this.state.logs !== undefined ? this.state.logs : []} />
                </div>
            </div>
        )
    }
}

const app = document.createElement('div');
document.body.appendChild(app);
ReactDOM.render(<Options />, app);