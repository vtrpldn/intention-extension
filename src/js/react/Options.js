import React from 'react'
import ReactDOM from 'react-dom'

const Options = () => {
  return (
    'Hi, I am the Options Page'
  )
}

const app = document.createElement('div')
document.body.appendChild(app)
ReactDOM.render(<Options />, app)

export default Options
