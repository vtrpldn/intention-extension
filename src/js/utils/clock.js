export const secondToClock = (seconds) => {
  let date = new Date(null)
  date.setSeconds(seconds)
  const timeString = date.toISOString().substr(11, 8)
  return timeString
}