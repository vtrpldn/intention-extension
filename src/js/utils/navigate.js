export const navigationTarget = {
  NEW_WINDOW: 'new-window',
  NEW_TAB: 'new-tab',
  CURRENT_TAB: 'current-tab'
}

export const navigate = (url, target = navigationTarget.NEW_TAB) => {
  switch (target) {
    case navigationTarget.NEW_WINDOW:
      return chrome.windows.create({ url, focused: true, type: 'normal' })
    case navigationTarget.CURRENT_TAB:
      return chrome.tabs.update({ url, active: true })
    default:
      return chrome.tabs.create({ url, active: true })
  }
}
