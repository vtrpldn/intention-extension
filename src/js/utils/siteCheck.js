export const getUrlListStatus = (list, url) => {
  console.log('list:', list)
  console.log('url:', url)
  if (list === []) {
    return false
  }
  return !!list.filter((v) => url.includes(v)).length
}

export const getUrlActiveStatus = (activeSites, url) => {
  console.log('activeSites:', activeSites)
  console.log('url:', url)
  if (activeSites === []) {
    return false
  }
  return !!activeSites.map(v => v.url).filter((v) => v === url).length
}
