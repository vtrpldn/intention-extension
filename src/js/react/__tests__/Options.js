import React from 'react'
import { shallow } from 'enzyme'
import Options from '../Options'
import '../../utils/enzyme'

describe('Options component testing with Enzyme', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Options />)
    expect(wrapper).toMatchSnapshot()
  })
})
