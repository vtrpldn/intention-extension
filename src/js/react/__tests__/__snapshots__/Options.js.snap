// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options component testing with Enzyme renders without crashing 1`] = `ShallowWrapper {}`;
